src/core/shape/ray.js

```diff
@@ -1,39 +1,43 @@
-import { Debug } from '../debug.js';
 import { Vec3 } from '../math/vec3.js';
 
 /**
  * An infinite ray.
  */
 class Ray {
+    /**
+     * The starting point of the ray.
+     *
+     * @type {Vec3}
+     */
+    origin = Vec3.ZERO.clone();
```

Comment:
we should add '@ignore' tag on the origin and direction members, to hide a public API allowing user to swap those to other instances of Vec3

Reply:
Thank you for your review @mvaligursky, I like your simplification! In this case, I am not sure, do you mean `@readonly`?

![image](https://user-images.githubusercontent.com/5236548/212660050-fd559f87-0222-4531-84a9-a51d84d0a270.png)

(it allows read-access while not just dropping it from docs)

Reply:
ah yeah, readonly is even better!

Reply:
Thank you a lot for the annotation hint! It feels much better now that Intellisense points it out.